asserts/database.go

```diff
@@ -125,26 +121,86 @@ func (db *Database) GenerateKey(authorityID string) (fingerprint string, err err
 		return "", fmt.Errorf("failed to generate private key: %v", err)
 	}
 
-	return db.ImportKey(authorityID, privKey)
+	return db.ImportKey(authorityID, WrapPrivateKey(privKey))
 }
 
 // ImportKey stores the given private/public key pair for identity and
 // returns its fingerprint
-func (db *Database) ImportKey(authorityID string, privKey *packet.PrivateKey) (fingerprint string, err error) {
-	buf := new(bytes.Buffer)
-	err = privKey.Serialize(buf)
+func (db *Database) ImportKey(authorityID string, privKey PrivateKey) (fingerprint string, err error) {
+	encoded, err := encodePrivateKey(privKey)
 	if err != nil {
 		return "", fmt.Errorf("failed to store private key: %v", err)
 	}
 
-	fingerp := hex.EncodeToString(privKey.PublicKey.Fingerprint[:])
-	err = db.atomicWriteEntry(buf.Bytes(), true, privateKeysRoot, authorityID, fingerp)
+	fingerp := privKey.PublicKey().Fingerprint()
+	err = db.atomicWriteEntry(encoded, true, privateKeysRoot, authorityID, fingerp)
 	if err != nil {
 		return "", fmt.Errorf("failed to store private key: %v", err)
 	}
 	return fingerp, nil
 }
 
+// findPrivateKey will return an error if not eactly one private key is found
+func (db *Database) findPrivateKey(authorityID, fingerprintWildcard string) (PrivateKey, error) {
+	keyPath := ""
+	foundPrivKeyCb := func(relpath string) error {
+		if keyPath != "" {
+			return fmt.Errorf("ambiguous search, more than one key pair found: %q and %q", keyPath, relpath)
+
+		}
+		keyPath = relpath
+		return nil
+	}
+	privKeysTop := filepath.Join(db.root, privateKeysRoot)
+	err := findWildcard(privKeysTop, []string{authorityID, fingerprintWildcard}, foundPrivKeyCb)
+	if err != nil {
+		return nil, err
+	}
+	if keyPath == "" {
+		return nil, fmt.Errorf("no matching key pair found")
+	}
+	encoded, err := db.readEntry(privateKeysRoot, keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read key pair: %v", err)
+	}
+	privKey, err := parsePrivateKey(encoded)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode key pair: %v", err)
+	}
+	return privKey, nil
+}
+
+// ExportPublicKey exports the public part of a stored key pair for identity
+// by matching the given fingerprint suffix, it is an error if no or more
+// than one key pair is found.
+func (db *Database) ExportPublicKey(authorityID string, fingerprintSuffix string) (PublicKey, error) {
+	privKey, err := db.findPrivateKey(authorityID, "*"+fingerprintSuffix)
+	if err != nil {
+		return nil, err
+	}
+	return privKey.PublicKey(), nil
+}
+
+// Sign makes an assertion with headers and body and signs it using the matching authority-id (from headers) key pair with the given fingeprint.
```

Comment:
Sorry, I'll nitpick further here, since the sentence as a whole feels hard to read.. let me propose something:
```
// Sign builds an assertion with the provided information and signs it.
// The assertion is signed with the private key from `headers["authority-id"]` that has the provided fingerprint.